stdlib/email/headerregistry.pyi

```diff
@@ -12,28 +14,34 @@ from email._header_value_parser import (
 )
 from email.errors import MessageDefect
 from email.policy import Policy
-from typing import Any, Iterable, Tuple, Type
+from typing import Any, ClassVar, Tuple, Type
+from typing_extensions import Literal
 
 class BaseHeader(str):
     @property
     def name(self) -> str: ...
     @property
     def defects(self) -> Tuple[MessageDefect, ...]: ...
     @property
+    @abstractmethod
     def max_count(self) -> int | None: ...
```

Comment:
Maybe this should be a `ClassVar`, not a property of each instance. Then it's no longer abstract, but it might be good if code like this worked:

```python3
def foo(header_class: Type[BaseHeader]) -> int | None:
    return header_class.max_count
```

Then again, the type annotation in this example is technically wrong, because you need a subclass of `BaseHeader`. But even then it wouldn't really work, because a typical header instance is a dynamically created class that multiple-inherits from `BaseHeader` and one of the other classes.

Reply:
I'm getting mixed messages here 😄 what we *really* want is an abstract class property, but I don't believe it's supported by mypy (and only works at runtime in >=3.9:

```
class BaseHeader:
    @classmethod
    @property
    @abstractmethod
    def max_count(cls) -> int | None: ...
```

I think your idea of changing it to a `ClassVar` for now is probably best, though. It's a lie, but a convenient lie.

Reply:
Neat -- switching it to a `ClassVar` apparently also means we can reduce the stubtest allowlist by 1 more, as well.